docs/tutorials/migrating-your-ga4-data-to-snowplow/index.md

```diff
@@ -0,0 +1,227 @@
+- `Start date`: The first date that you want to query data from. This is inclusive.
+- `End date`: The date that you want to query data until. This is inclusive. If you only want to query only one date, set both start and end dates to that date.
+
+### Step 2 - Map Snowplow events
+
+Although GA4 has events, it does not have a concept of entities like Snowplow. As such, we gather the parameters from each of your GA4 events, and add similar parameters to a Snowplow entity that we create. For example, if you have two events, `add_to_cart` and `view_cart`, with each containing a property called `page_title`, we will create a Snowplow entity called context_page_vendor_1_0_0, and add the `page_title` parameter to it. In addition, for each of your events, a Snowplow unstruct event will also be created, containing the parameters from each GA4 event.\
+
+The interface in the UI allows you to modify any of the pre-created entities, or create your own entities. You can remove parameters the events (for example if you wish to add the parameter to an entity), but you cannot add any new parameters to events. Events will be migrated contatining the same parameters as they had in GA4, minus any that you remove.\
+When you create a new entity, you can choose parameters from the events that are shown in the UI. Note that you only have to add a parameter to an entity once, and it will be filled in the entity column from any event that has that parameter.\
+For example, if you move the `page_title` parameter (which exists in both `add_to_cart` and `view_cart` events) from the `add_to_cart` event to a new `cart` entity, in the newly created table the `page_title` parameter will be filled in on both the `add_to_cart` and `view_cart` events. You don't have to copy the parameter from both events to the new entity.\
+
+*Note: When naming new entities, only use alphanumeric characters and underscores. Do not use spaces or special characters.*
+
+### Step 3 - Map additional columns
+
+In this step, we are creating the mappings from GA4 to the native Snowplow columns. There are defaults that are set, but these can be changed if you wish.\
+`app_id`, for example, must be changed, substituting the placeholder values for the correct values for your GA4 data.\
+The SQL that you input must conform to strict requirements:
+- The SQL written for the mapping will be placed directly into the code, so it must be valid as a standalone single-line statement:
+```sql
+...
+<your input SQL> AS snowplow_field
+...
+```
+| Snowplow_field | mapping_sql |
+| --------------- | ----------- |
+| page_urlscheme  | regexp_extract(page_url, r'(.*?):') |
+| page_urlhost    | {{dbt_utils.get_url_host(field='page_url')}} |
+| geo_zipcode     | NULL        |
+| app_id          | CASE WHEN stream_id = '4271243942' THEN 'Snowplow' ELSE NULL END |
+
+For example, at compile time, the above will look like:
```

Comment:
hmmm yeah I think what i've written is wrong anyway - it doesn't look like that when dbt compiles because at compile time it will evaluate the curly braces. I'll reword